libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -1027,3 +1060,347 @@ void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
      * from the server to complete the incoming PUBLISH QoS2 transaction. */
     TEST_ASSERT_EQUAL( MQTT_PACKET_ID_INVALID, context.incomingPublishRecords[ 0 ].packetId );
 }
+
+/**
+ * @brief Verifies the behavior of the MQTT library in a restored session with the broker
+ * for resending QoS 1 PUBLISH packets that were un-acknowledged in the previous connection.
+ * Tests that the library is able to support resending the PUBLISH packet with the DUP flag
+ * set in the restored session.
+ */
+void test_MQTT_Restore_Session_Resend_Unacked_Publish_QoS1( void )
```

Comment:
I think this test should be called something more along the lines of `test_MQTT_PublishToResend_QoS1`, since MQTT_PublishToResend is really the only part of the library we're testing; the actual resending is done by the application

Reply:
We are also testing that the state transitions of the records are occurring correctly when resending the PUBLISH packet ID

Reply:
Thinking about it, I agree with your point of not tying the test of resending unacked PUBLISH packet with persistent session in the test name. Have changed the name of both PUBLISH resend tests